app/jobs/regular/problem_check.rb

```diff
@@ -18,7 +18,12 @@ def execute(args)
       problems = check.call
       raise RetrySignal if problems.present? && retry_count < check.max_retries
 
-      problems.each { |problem| AdminDashboardData.add_found_scheduled_check_problem(problem) }
+      if problems.present?
+        problems.each { |problem| AdminDashboardData.add_found_scheduled_check_problem(problem) }
+        ProblemCheckTracker[identifier].problem!(next_run_at: check.perform_every.from_now)
+      else
+        ProblemCheckTracker[identifier].no_problem!(next_run_at: check.perform_every.from_now)
```

Comment:
We may also want `clear_found_problem` to call this (it's the opposite of `add_found_scheduled_check_problem`). For example, in the Group email UI, you have to test the credentials before saving, so when you do save we clear the problem because we know it's resolved:

https://github.com/discourse/discourse/blob/1bd9ca11e777f880462ae64c6795ef7de28a8cd2/app/controllers/groups_controller.rb#L175-L182